pkg/networkpath/traceroute/runner.go

```diff
@@ -127,36 +183,60 @@ func processResults(r *results.Results, hname string, destinationHost string, de
 		PathID:    pathID,
 		Timestamp: time.Now().UnixMilli(),
 		Source: NetworkPathSource{
-			Hostname: hname,
+			Hostname:  hname,
+			NetworkID: r.networkID,
 		},
 		Destination: NetworkPathDestination{
 			Hostname:  destinationHost,
 			IPAddress: destinationIP.String(),
 		},
 	}
 
-	for idx, probes := range r.Flows {
+	for idx, probes := range res.Flows {
 		log.Debugf("Flow idx: %d\n", idx)
 		for probleIndex, probe := range probes {
 			log.Debugf("%d - %d - %s\n", probleIndex, probe.Sent.IP.TTL, probe.Name)
 		}
 	}
 
-	flowIDs := make([]int, 0, len(r.Flows))
-	for flowID := range r.Flows {
+	flowIDs := make([]int, 0, len(res.Flows))
+	for flowID := range res.Flows {
 		flowIDs = append(flowIDs, int(flowID))
 	}
 	sort.Ints(flowIDs)
 
 	for _, flowID := range flowIDs {
-		hops := r.Flows[uint16(flowID)]
+		hops := res.Flows[uint16(flowID)]
 		if len(hops) == 0 {
 			log.Debugf("No hops for flow ID %d", flowID)
 			continue
 		}
 		var nodes []node
 		// add first hop
-		firstNodeName := hops[0].Sent.IP.SrcIP.String()
+		localAddr := hops[0].Sent.IP.SrcIP
+
+		// get hardware interface info
+		iface, err := getInterface(localAddr)
+		if err != nil {
+			// TODO: we probably don't want to stop execution here.
+			// For testing though, we probably do want to see how often
+			// this fails
+			log.Errorf("failed to get interface: %s", err.Error())
+			return NetworkPath{}, err
+		}
+
+		// Resolve subnet from hardware interface
+		// we should probably cache this in a similar
+		// way to what's done for NPM
+		subnet, err := r.resolveSubnetID(ctx, iface)
+		if err != nil {
+			log.Errorf("failed to resolve subnet for interface %q: %s", iface.Name, err.Error())
+		}
```

Comment:
I think it can help maintainability/readability to extract the subnet resolution into a different file/class. Something like "SubnetResolver" that includes:
- the logic for resolving addr -> interface -> subnet id
- the various telemetry data